src/lib/ifaces/mac_vtap.rs

```diff
@@ -0,0 +1,92 @@
+use crate::ifaces::mac_vlan::get_mac_vlan_info;
+use crate::ifaces::mac_vlan::MacVlanInfo;
+use crate::ifaces::mac_vlan::MacVlanMode;
+use netlink_packet_route::rtnl::link::nlas;
+use serde_derive::{Deserialize, Serialize};
+
+/* mac_vlan is sharing the same code with mac_vtap in kernel for netlink */
+const MACVLAN_MODE_PRIVATE: u32 = 1;
+const MACVLAN_MODE_VEPA: u32 = 2;
+const MACVLAN_MODE_BRIDGE: u32 = 4;
+const MACVLAN_MODE_PASSTHRU: u32 = 8;
+const MACVLAN_MODE_SOURCE: u32 = 16;
+
+#[serde(rename_all = "lowercase")]
+#[derive(Serialize, Deserialize, Debug, PartialEq, Clone)]
+pub enum MacVtapMode {
+    /* don't talk to other macvlans */
+    Private,
+    /* talk to other ports through ext bridge */
+    Vepa,
+    /* talk to bridge ports directly */
+    Bridge,
+    /* take over the underlying device */
+    #[serde(rename = "passthru")]
+    PassThrough,
+    /* use source MAC address list to assign */
+    Source,
+    Other(u32),
+    Unknown,
+}
+
+impl Default for MacVtapMode {
+    fn default() -> Self {
+        MacVtapMode::Unknown
+    }
+}
+
+impl From<u32> for MacVtapMode {
```

Comment:
This is not required and the constants it used.

Reply:
Done, thanks!